File: Src/ButterflyNet.Parser/Satisfiers/ExactCharMatchSatisfier.cs
namespace ButterflyNet.Parser.Strategies {
	public class ExactCharMatchSatisfier : ISatisfier {
		private readonly string chars;

		public ExactCharMatchSatisfier(string chars) {
			this.chars = chars;
		}

		public bool IsSatisfiedBy(ParseContext context) {
			if (chars.Length == 1) {
				return chars[0] == context.Input.Current;
			}

			return (char)context.Input.Current + context.Input.Peek(chars.Length - 1) == chars;
		}
	}
}

File: src/js/src/satisfiers/ExactCharMatchSatisfier.js
function ExactCharMatchSatisfier(chars) {
	
	this.isSatisfiedBy = function(context) {
		if (chars.length === 1) {
			return chars === context.currentChar;
		}
		
		return context.currentChar + context.input.peek(chars.length - 1) === chars;
	};
}